tests/test_water.py

```diff
@@ -82,14 +95,21 @@ def test_water_sound_speed():
                        1555.76629165, 1555.55766354, 1555.31270576, 1555.03176554,
                        1554.71518402, 1554.3632965, 1553.97643281, 1553.55491766,
                        1553.09907095, 1552.60920812, 1552.08564046, 1551.52867549,
-                       1550.93861723, 1550.31576661, 1549.66042173, 1548.97287827]
+                       1550.93861723, 1550.31576661, 1549.66042173, 1548.97287827])
+    input_temp = np.arange(96, dtype=float)
 
     for idx, value in enumerate(expected_values):
         assert abs(value - water_sound_speed(idx)) < 1e-6, "Expected value deviates from water_sound_speed output"
+    
+    assert np.max(np.abs(expected_values - water_sound_speed(input_temp))) < 1e-6, "Expected value deviates from water_sound_speed output"
+    assert np.max(np.abs(expected_values.reshape(12,8) - water_sound_speed(input_temp.reshape(12,8)))) < 1e-6, "Expected value deviates from water_sound_speed output"
```

Comment:
These two tests look similar but their input data shape is different. If I understand this right, you're testing function under different shapes. Maybe update the error messages to reflect the test type/goal? This will help with quickly identifying the error type when we have one.